src/mode/mode_struct.rs

```diff
@@ -1215,10 +1270,43 @@ impl<T: Transformation> Mode<T> {
         } else {
             pepped_up_control_value.calc_distance_from(current_target_value.to_continuous_value())
         };
+
+        // If there is no previous pepped-up control value, set one
+        let previous_pepped_up_control_value = match self.state.previous_pepped_up_control_value {
```

Comment:
You can shorten this block thanks to the nice method `replace` of `Option`:

```suggestion
        let previous_pepped_up_control_value = self.state.previous_pepped_up_control_value
            .replace(pepped_up_control_value)
            .unwrap_or(pepped_up_control_value);
```

BTW, is it really your intention to set the `previous_pepped_up_control_value` to the current value at the first time?